### ISSUE TITLE
Missing notifications and new messages previews in the contact list 

### ISSUE THREAD

**Issue by abergmot** [movim/movim#1318]
After [this other issue was solved](https://github.com/movim/movim/issues/1317), a problem is occuring with new messages in private 1:1 chats : Even though we get the global notifications in the left bar that a new private message is available, when clicking on the icon to show the contact list, nothing is updated or highlighted. So we have to guess who wrote to us, and then, by clicking on the contact, the new message then appears inside the newly opened chat window.

### Edit : simpler picture to understand
Here's a screenshot of the notifications on two different instances. The left one is hidding some, when the picture on the right shows full notifications on another instance (I fully logged out from the first to log in into the second instance).

![notifs](https://github.com/movim/movim/assets/34137319/f8f79c00-fa4e-44a9-8f7c-0dddcc4fbea3)

....

On a rare occasion, I don't know why, the preview of the new message did appear in the contact list, but nothing was highlighting it so if you have a big contact list you still have to guess which one of your contacts wrote a new message to you, because the notification icon showing on the contacts wasn't there anymore. 
![no-notif-2b](https://github.com/movim/movim/assets/34137319/f529f1ab-df48-4c07-a90d-bcc4c352a00c)

(For this tests, all my extensions are disabled. I'm using Firefox 125.0.3 but my friends use different versions which are up to date or almost so.)

This lack of preview and precise notifications **is preventing people to get new messages**, because the global notification seems bogus or related to something else, and **you have to manually check every contact to update the content of the chats**.

-----

### New tests to replicate

**24th of May** : The issue never disappeared for me and my friends in the exact same conditions described in this post. After more tests, I can confirm that the exact issue is still occurring in different places, with different service providers, on different devices and browsers. 
On the contrary, when using Pidgin or Dino clients instead of https://mov.im , at my place, the new messages show instantly where they should, no issue.

---

**Tested and reproduced** at my place on :
- old firefox (91.8.0esr)
- Chromium Browser (stable)
- Torbrowser

On other people's devices and network :
- On Fennec android browser
- On movim PWA app (same android device as previuous one)

I'll add more with screenshots when I can get them.

----
### INSTANCES 
**EDIT 2:** It's not showing the same on all instances : 
Loging in some different instance (version 0.24.1) shows the new message as it should (here on https://movim.teftera.com/)
![movim teftera com-Screenshot 2024-05-24 at 12-46-34 Discussions](https://github.com/movim/movim/assets/34137319/91b0ad72-b261-48af-80c9-1c018ee5e40b)
This above should is the normal behaviour of notifications that's broken on https://mov.im

Back on https://mov.im the unread message still isn't notified and previewd in contact list.

Notifications don't show either on this instances: https://xmpp.earth (version 0.24.1) : 
![xmpp earth-19-31_2Screenshot 2024-05-17 at 19-37-31 Actualité](https://github.com/movim/movim/assets/34137319/e5e3c6da-7b91-4bc0-bdc7-64cf85465db5)

-----

### IMPORTANT : ISSUE NOT OCCURING WITH ONLINE CONTACTS (both logged in)
**EDIT 3** : **The issue is not occuring** when me and my contacts are online (logged in) at the same time on https://mov.im, and at that point 1:1 chat notifications + message previews do appear in contact list before I open the chat, as they should. Now if I log out, and someone send me a message while I'm not online, the chat notification won't get through when I log in later while my contact is also logged out, as shown on the first picture.

**Comment by edhelas** [comment]
So let met clarify something. Is this issue occurring SINCE #1317 or was is there before ?

**Comment by abergmot** [comment]
> So let met clarify something. Is this issue occurring SINCE #1317 or was is there before ?

Before #1317, it was impossible to see a message at all, even by opening the chat window with the person who had sent one. I had no global notifications either.

Now I get the global notification on the left, but nothing else, no preview of the message in contact list. But if I know who wrote to me and I open a window with that contact, then I can read the message, unlike before.

**Comment by edhelas** [comment]
And when you open the chat, does the notification counter actually clear ?

**Comment by abergmot** [comment]
yes, the global notification in the left bar is cleared when I open the right chat, and then a short preview of the message also appears in the contact list where it should have been in the first place.

**Comment by abergmot** [comment]
I updated with more tests verifying that new chats are not showing on contacts list in https://mov.im
The issue is occuring on multiple devices/browsers/apps and at least on two different service providers. A third one is yet to be confirmed, and I'll try to post screenshots to assess that.

**Comment by edhelas** [comment]
Hi @abergmot , is it still happening on your side, I made some code change related to that part and was wondering if its still happening ?

**Comment by ekucks21** [comment]
I also am seeing this or a similar behavior. I can see the new messages come in on cheogram but I have to manually add a chatroom in movim.chatterboxtown.us before the messages will show up.

**Comment by edhelas** [comment]
I did heavy refactoring around that topic, so far I didn't experienced the issue anymore.
I'm closing the ticket for now. If you find a proper way to reproduce it I can reopen, and try to fix, the ticket :)
